deeplearning4j-scaleout/spark/dl4j-spark/src/main/java/org/deeplearning4j/spark/impl/paramavg/BaseTrainingMaster.java

```diff
@@ -0,0 +1,247 @@
+import org.nd4j.shade.jackson.annotation.PropertyAccessor;
+import org.nd4j.shade.jackson.core.JsonFactory;
+import org.nd4j.shade.jackson.databind.DeserializationFeature;
+import org.nd4j.shade.jackson.databind.MapperFeature;
+import org.nd4j.shade.jackson.databind.ObjectMapper;
+import org.nd4j.shade.jackson.databind.SerializationFeature;
+import org.nd4j.shade.jackson.databind.annotation.JsonDeserialize;
+import org.nd4j.shade.jackson.databind.annotation.JsonSerialize;
+import org.nd4j.shade.jackson.dataformat.yaml.YAMLFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Random;
+
+/**
+ * @author raver119@gmail.com
+ * @author Alex Black
+ */
+@Slf4j
+public abstract class BaseTrainingMaster<R extends TrainingResult, W extends TrainingWorker<R>> implements TrainingMaster<R, W> {
+    protected static ObjectMapper jsonMapper;
+    protected static ObjectMapper yamlMapper;
+
+    protected boolean collectTrainingStats;
+    protected ParameterAveragingTrainingMasterStats.ParameterAveragingTrainingMasterStatsHelper stats;
+
+    protected int lastExportedRDDId = Integer.MIN_VALUE;
+    protected String lastRDDExportPath;
+    protected int batchSizePerWorker;
+    protected String exportDirectory = null;
+    protected Random rng;
+
+    protected String trainingMasterUID;
+
+
+    protected Repartition repartition;
+    protected RepartitionStrategy repartitionStrategy;
+    @JsonSerialize(using = StorageLevelSerializer.class)
+    @JsonDeserialize(using = StorageLevelDeserializer.class)
+    protected StorageLevel storageLevel;
+    @JsonSerialize(using = StorageLevelSerializer.class)
+    @JsonDeserialize(using = StorageLevelDeserializer.class)
+    protected StorageLevel storageLevelStreams = StorageLevel.MEMORY_ONLY();
+    protected RDDTrainingApproach rddTrainingApproach = RDDTrainingApproach.Export;
+
+    protected BaseTrainingMaster() {
+
+    }
+
+
+    protected static synchronized ObjectMapper getJsonMapper() {
+        if (jsonMapper == null) {
+            jsonMapper = getNewMapper(new JsonFactory());
+        }
+        return jsonMapper;
+    }
+
+    protected static synchronized ObjectMapper getYamlMapper() {
+        if (yamlMapper == null) {
+            yamlMapper = getNewMapper(new YAMLFactory());
+        }
+        return yamlMapper;
+    }
+
+    protected static ObjectMapper getNewMapper(JsonFactory jsonFactory) {
+        ObjectMapper om = new ObjectMapper(jsonFactory);
+        om.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        om.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+        om.configure(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY, true);
+        om.enable(SerializationFeature.INDENT_OUTPUT);
+        om.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE);
+        om.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+        return om;
```

Comment:
I realize this is just copied over from the original impl - but these JSON mapper get/config methods - I'm thinking they are probably reusable elsewhere... maybe move them to a util class so we can use them elsewhere as needed?